it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,362 @@
+        this.projectId = projectId;
+        this.createdStreamIds = Collections.synchronizedSet(new HashSet<>());
+        this.createdConnectionProfileIds = Collections.synchronizedSet(new HashSet<>());
+    }
+
+    public static Builder builder(String projectId, String location) {
+        return new Builder(projectId, location);
+    }
+
+    public ConnectionProfile createJDBCSourceConnectionProfile(String connectionProfileId, JDBCSource source) {
+
+        LOG.info("Creating JDBC Source Connection Profile {} in project {}.", connectionProfileId, projectId);
+
+        ConnectionProfile.Builder connectionProfileBuilder = ConnectionProfile.newBuilder();
+
+        try {
+            switch (source.getType()) {
+                case MYSQL:
+                    MysqlProfile.Builder mysqlProfileBuilder = MysqlProfile.newBuilder();
+                    mysqlProfileBuilder.setHostname(source.getHostname()).setUsername(source.getUsername()).setPassword(source.getPassword()).setPort(source.getPort());
+                    connectionProfileBuilder.setMysqlProfile(mysqlProfileBuilder);
+                case ORACLE:
+                    OracleProfile.Builder oracleProfileBuilder = OracleProfile.newBuilder();
+                    oracleProfileBuilder.setHostname(source.getHostname()).setUsername(source.getUsername()).setPassword(source.getPassword()).setPort(source.getPort());
+                    connectionProfileBuilder.setOracleProfile(oracleProfileBuilder);
+                case POSTGRESQL:
+                    PostgresqlProfile.Builder postgresqlProfileBuilder = PostgresqlProfile.newBuilder();
+                    postgresqlProfileBuilder.setHostname(source.getHostname()).setUsername(source.getUsername()).setPassword(source.getPassword())
+                            .setPort(source.getPort()).setDatabase(((PostgresSource) source).getDatabase());
+                    connectionProfileBuilder.setPostgresqlProfile(postgresqlProfileBuilder);
+            }
+
+            ConnectionProfile connectionProfile = connectionProfileBuilder
+                    .setDisplayName(connectionProfileId)
+                    .setStaticServiceIpConnectivity(StaticServiceIpConnectivity.getDefaultInstance())
+                    .build();
+            CreateConnectionProfileRequest request = CreateConnectionProfileRequest.newBuilder()
+                    .setParent(LocationName.of(projectId, location).toString())
+                    .setConnectionProfile(connectionProfile)
+                    .setConnectionProfileId(connectionProfileId)
+                    .build();
+            ConnectionProfile reference = datastreamClient.createConnectionProfileAsync(request).get();
+            createdConnectionProfileIds.add(connectionProfileId);
+
+            return reference;
+        } catch (ExecutionException | InterruptedException e) {
+            throw new DatastreamResourceManagerException("Failed to create JDBC source connection profile. ", e);
+        }
+
```

Comment:
nit: Could use a "...generated connection profile ... successfully..." log at the end.